modules/integration-core/src/test/scala/scala/cli/integration/NativePackagerTests.scala

```diff
@@ -0,0 +1,74 @@
+package scala.cli.integration
+
+import com.eed3si9n.expecty.Expecty.expect
+
+import scala.util.Properties
+
+class NativePackagerTests extends munit.FunSuite{
+
+  val helloWorldFileName = "HelloWorldScalaCli.scala"
+  val message = "Hello, world!"
+  val helloWorldTestInputs = TestInputs(
+    Seq(
+      os.rel / helloWorldFileName ->
+        s"""
+          |object HelloWorld {
+          |  def main(args: Array[String]): Unit = {
+          |    println("$message")
+          |  }
+          |}""".stripMargin
+    )
+  )
+
+  if (Properties.isMac) {
+    test("building pkg package") {
+
+      helloWorldTestInputs.fromRoot { root =>
+
+        val appName = helloWorldFileName.stripSuffix(".scala").toLowerCase
+        val pkgAppFile = s"$appName.pkg"
+        os.proc(TestUtil.cli, "package", helloWorldFileName, "--pkg", "--output-package-path", pkgAppFile).call(
+          cwd = root,
+          stdin = os.Inherit,
+          stdout = os.Inherit,
+          stderr = os.Inherit
+        )
+
+        val pkgAppPath = root / pkgAppFile
+        expect(os.isFile(pkgAppPath))
+
+        if( TestUtil.isCI) {
+          os.proc("installer", "-pkg", pkgAppFile, "-target", "CurrentUserHomeDirectory").call(
+            cwd = root,
+            stdin = os.Inherit,
+            stdout = os.Inherit,
+            stderr = os.Inherit
+          )
+
+          val output = os.proc(s"/Users/runner/Applications/$appName.app/Contents/MacOS/$appName").call(cwd = os.root).out.text.trim
```

Comment:
Just in case, we could get the home directory, rather than hard-coding it:
```suggestion
          val home = sys.props("user.home")
          val output = os.proc(s"$home/Applications/$appName.app/Contents/MacOS/$appName").call(cwd = os.root).out.text.trim
```